src/Sentry/Threading/ScopedCountdownLock.cs

```diff
@@ -0,0 +1,149 @@
+namespace Sentry.Threading;
+
+/// <summary>
+/// A synchronization primitive that tracks the amount of <see cref="CounterScope"/>s held,
+/// and is signaled when the count reaches zero while a <see cref="LockScope"/> is held.
```

Comment:
> is signaled when the count reaches zero while a `LockScope` is held

Is this accurate? Looking at the code it signals during `ExitCounterScope`:
https://github.com/getsentry/sentry-dotnet/blob/e8850db578a5f3a4ba9afb940fa934e365e89020/src/Sentry/Threading/ScopedCountdownLock.cs#L59

... which is only called when the lockObj is not null:
https://github.com/getsentry/sentry-dotnet/blob/e8850db578a5f3a4ba9afb940fa934e365e89020/src/Sentry/Threading/ScopedCountdownLock.cs#L110-L114

And that only happens if the CountdownEvent is **not zero**:
https://github.com/getsentry/sentry-dotnet/blob/e8850db578a5f3a4ba9afb940fa934e365e89020/src/Sentry/Threading/ScopedCountdownLock.cs#L49-L52

So it seems it's signalled when releasing a lock scope and when the count has not yet reached zero.